pydra/engine/tests/test_task.py

```diff
@@ -1361,3 +1361,33 @@ def fun_error(x):
     # the error traceback should be a list and should point to a specific line in the function
     assert isinstance(error_tb, list)
     assert "in fun_error" in error_tb[-2]
+
+
+def test_rerun_errored(tmpdir, capfd):
+    @mark.task
+    def pass_odds(x):
+        if x % 2 == 0:
+            print(f"x%2 = {x % 2} (error)")
+            raise Exception("even error")
+        else:
+            print(f"x%2 = {x % 2}")
+            return x
+
+    task = pass_odds(name="pass_odds", x=[1, 2, 3, 4, 5], cache_dir=tmpdir).split("x")
+
+    task()
+
+    with pytest.raises(Exception, match="even error") as exinfo:
+        task()
+    print("...")
```

Comment:
remove print.

Reply:
I kept the print("...") just to make a distinction in the stdout of running the task the first time versus the second time. print("...") could be removed, but I based the assert statements on stdout, so the assert statements would have to be adjusted.

Reply:
if a print is needed, a logger instead of print so that it gets captured better by pytest.